pkg/registration/spoke/bootstrapkubeconfigsmanager/bootstrapkubeconfigsmanager.go

```diff
@@ -0,0 +1,122 @@
+package bootstrapkubeconfigsmanager
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// bootstrapkubeconfigsManager manages multiple bootstrapkubeconfigs.
+// The only public method is ReSelect.
+// ReSelect do 2 things:
+// 1. Call Fail method of the bootstrapkubeconfig that has the same kubeconfig data with the current bootstrapkubeconfig in use.
+// 2. Select a first valid bootstrapkubeconfig from the list and update it to the current bootstrapkubeconfig in use.
+//
+// Note: ReSelect should be called only once. Then it waits for registration-agent to restart and rebootstrap.
```

Comment:
Need another controller to sync changes in secret in the list to bootstarap-hub-kubeconfig when use actively change it.

Reply:
Done, another controller added.